src/Component/Layout/Header.tsx

```diff
@@ -1,24 +1,40 @@
 import styled from '@emotion/styled';
+import { Modal } from '../Modal';
+import CartItemModalContent from '../Modal/CartItemModalContent';
+import { CartItemTypes } from '../../types/CartItemType';
+import getShoppingCart from '../../api/getShoppingCart';
+import { useAPIContext } from '../Common/Provider';
 
 interface HeaderProps {
-  cartItemCount: number;
-  status: 'idle' | 'loading' | 'success' | 'error';
+  updateErrorMessage: (errorMessage: string) => void;
 }
 
-export default function Header({ cartItemCount, status }: HeaderProps) {
+export default function Header({ updateErrorMessage }: HeaderProps) {
+  const { data: cartItems, status } = useAPIContext<CartItemTypes[]>({
+    apiFn: () => getShoppingCart(),
+    key: 'cartItems',
+  });
```

Comment:
다음과 같이 하나의 훅으로 일원화하여 관리할 수 있을 것으로 생각됩니다.

```suggestion
  const useCartItems = () => useAPIContext<CartItemTypes[]>({
    apiFn: () => getShoppingCart(),
    key: 'cartItems',
  })
  
  ...
  
  import useCartItems from '~~';

  const { data: cartItems, status } = useCartItems();
```

Reply:
수정 커밋: [refactor: useCartItems 훅을 이용해 useAPIContext 사용하던 부분 변경](https://github.com/woowacourse/react-shopping-products/pull/153/commits/63733c12dc1ceda8c4140d0bbe5760908f5acdb0)